### PR TITLE
feat: take into count of multi-width characters on offset of highlights and jump

### PR DESCRIPTION
**Pull request by atusy**
solves #121 

I added `util.offset_pos()` and used it to calculate correct offset.
If you want to relocate or rename it, give me a comment or feel free to do it!

### CODE REVIEW

**Comment by atusy** [comment]
converting draft as operator pending mode is giving wrong offset

**Comment by atusy** [comment]
should be okay now

File: lua/flash/util.lua [hunk 1]
```diff
@@ -13,6 +13,30 @@ M.BS = M.t("<bs>")
...
+function M.offset_pos(buf, pos, offset)
+  local row = pos[1] - 1 + offset[1]
+  local ok, lines = pcall(vim.api.nvim_buf_get_lines, buf, row, row + 1, true)
+  if not ok or lines == nil then
+    -- fallback to old behavior if anything wrong happens
+    return {row + 1, math.max(pos[2] + offset[2], 0)}
+  end
+
+  local line = lines[1]
+  local charidx = vim.fn.charidx(line, pos[2])
```
Comment (folke) [inline lua/flash/util.lua:29]: Can't we just do this here?

```lua
local col = vim.fn.byteidx(line, charidx + offset[2])
return {row + 1, math.max(col + coloffset, 0)}

Comment (atusy) [reply lua/flash/util.lua:29]: oh it seems it does!! cool!!!

**Comment by folke** [comment]
thanks!

**Pull request merged**